data_handler/handlers/loan_states/hashtack_v0/interest_rate.py

```diff
@@ -0,0 +1,87 @@
+from decimal import Decimal
+
+from db.models import InterestRate
+from handler_tools.api_connector import DeRiskAPIConnector
+from handlers.loan_states.hashtack_v0.events import HashstackV0Event
+
+HASHSTACK_INTEREST_RATE_ADDRESS = "0x01b862c518939339b950d0d21a3d4cc8ead102d6270850ac8544636e558fab68"
+
+SECONDS_IN_YEAR = Decimal(365 * 24 * 60 * 60)
+
+
+class HashstackV0InterestRate:
+    """Class for calculating interest rates for a given token on the Hashstack V0 protocol."""
+
+    def __init__(self, token: str, min_block_number: int, max_block_number: int):
+        """
+        Initialize the HashstackV0InterestRate object.
+        :param token: str - The token address in hexadecimal.
+        :param min_block_number: int - The lower block number from which to retrieve events.
+        :param max_block_number: int - The upper block number to which to retrieve events.
+        """
+        self.token = token
+        self.min_block_number = min_block_number
+        self.max_block_number = max_block_number
+        self.interest_rates = []
+        self.api_connector = DeRiskAPIConnector()
+        self.events = None
+        self._set_events()
+
+    def _set_events(self):
+        """Fetch events from the API, filter them by token and set them to the events attribute."""
+        if not isinstance(self.min_block_number, int) or not isinstance(self.max_block_number, int):
+            raise ValueError("Block numbers must be integers")
+        result = self.api_connector.get_data(
+            HASHSTACK_INTEREST_RATE_ADDRESS,
+            self.min_block_number,
+            self.max_block_number,
+        )
+        if isinstance(result, dict):
+            raise RuntimeError(f"Error while fetching events: {result.get('error', 'Unknown error')}")
+        self.events = list(filter(lambda event: event["data"][0] == self.token, result))
+
+    def calculate_interest_rates(self) -> None:
+        """Calculate the interest rates for the given token."""
+        if not self.events:
+            return
+        supply_interest_rate_cum = Decimal("0")
+        borrow_interest_rate_cum = Decimal("0")
+        percents_decimals_shift = Decimal(10000)
```

Comment:
```suggestion
        BASIS_POINT = Decimal("0.0001")
```
.. And then use it to multiply `borrow_apr` and `supply_apr`.